src/components/search/search.component.tsx

```diff
@@ -245,7 +245,7 @@ export const Search = React.forwardRef<SearchHandle, SearchProps>(
         variant={variant}
         mb={0}
         {...filterStyledSystemMarginProps(rest)}
-        {...tagComponent("search", rest)}
+        {...tagComponent("search", { "data-role": "search", ...rest })}
```

Comment:
**suggestion(non-blocking)**: what do you think about defining the props directly here? It might help with readability.

```suggestion
        data-component="search"
        data-role="search"
```